userland/configure.in

```diff
@@ -5,15 +5,15 @@ AC_CONFIG_FILES(lib/Makefile)
 
 AC_CHECK_HEADERS([linux/net_tstamp.h librdi.h])
 
-MACHINE=`uname -m`
 CFLAGS=""
 SYS_LIBS=""
 
 VER=`cat ../kernel/linux/pf_ring.h | grep RING_VERSION | head -1 | cut -d '"' -f 2`
 MAJOR_VER=`cat ../kernel/linux/pf_ring.h | grep RING_VERSION | head -1 | cut -d '"' -f 2 | cut -d '.' -f 1`
 
 dnl> Checking for supported libraries (corei7, corei7-avx, core-avx2)
-NATIVE=`$CC -c -Q -march=native --help=target| grep "march" | xargs | cut -d ' ' -f 2`
+NATIVE=`$CC -c -Q --help=target | tr -d '[[:blank:]]' | grep "^\-march\=" | cut -d '=' -f 2`
```

Comment:
Hi,
how can I reproduce the compilation failure?
I modified that line, so that the variable NATIVE is set to the (cross)compiler target architecture: x86-64 / armv2 / whatever.  
I then, modified MACHINE=$NATIVE and I think that can be the cause of what you're seing.

If the problem is only the NATIVE variable, we could try the folloring:

NATIVE=`$CC -c -Q -march=native --help=target| grep "march" | xargs | cut -d ' ' -f 2`
MACHINE=`$CC -c -Q --help=target | tr -d '[[:blank:]]' | grep "^\-march\=" | cut -d '=' -f 2`

this way, machine gets the name of the target architecture and crosscompilation works.
I've tested it with a few different toolchains with buildroot

Reply:
I've created a branch withe the reworked patch: dev-improve-crosscompilation-v3

any chance you cat try it?

thank you
Davide

Reply:
Yes your patch should be fine, please send a pull request for the new code. Thank you.